lighthouse-core/test/lib/network-request-test.js

```diff
@@ -176,5 +183,125 @@ describe('NetworkRequest', () => {
+      expect(record.timing.sendStart).toStrictEqual(0);
+      expect(record.timing.sendEnd).toStrictEqual(0);
+      expect(record.timing.receiveHeadersEnd).toStrictEqual(0);
+      expect(record.lrStatistics).toStrictEqual({
+        endTimeDeltaMs: -8000,
+        TCPMs: 0,
+        requestMs: 0,
+        responseMs: 10000,
+      });
+    });
+
+    it('treats missing timings as 0', function() {
+      const req = getRequest();
+      req.responseHeaders = [{name: NetworkRequest.HEADER_TOTAL, value: '10000'},
+        {name: NetworkRequest.HEADER_TCP, value: '1000'},
+        {name: NetworkRequest.HEADER_RES, value: '9000'}];
+
+      const devtoolsLog = networkRecordsToDevtoolsLog([req]);
+      global.isLightrider = true;
+      const record = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
+
+      expect(record.startTime).toStrictEqual(0);
+      expect(record.endTime).toStrictEqual(10);
+      expect(record.responseReceivedTime).toStrictEqual(1);
+      expect(record.timing.connectStart).toStrictEqual(0);
+      expect(record.timing.connectEnd).toStrictEqual(1000);
+      expect(record.timing.sslStart).toStrictEqual(1000);
+      expect(record.timing.sslEnd).toStrictEqual(1000);
+      expect(record.timing.sendStart).toStrictEqual(1000);
+      expect(record.timing.sendEnd).toStrictEqual(1000);
+      expect(record.timing.receiveHeadersEnd).toStrictEqual(1000);
+      expect(record.lrStatistics).toStrictEqual({
+        endTimeDeltaMs: -8000,
+        TCPMs: 1000,
+        requestMs: 0,
+        responseMs: 9000,
+      });
+    });
+
+    it('created a new timing property if one did not exist', function() {
+      const req = getRequest();
+
+      const devtoolsLog = networkRecordsToDevtoolsLog([req]);
+
+      const noLRRecord = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
+      expect(noLRRecord.timing).toStrictEqual(undefined);
+
+      global.isLightrider = true;
+      const lrRecord = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
+      expect(lrRecord.timing.proxyStart).toStrictEqual(-1);
+
+      expect(lrRecord.startTime).toStrictEqual(0);
+      expect(lrRecord.endTime).toStrictEqual(10);
+      expect(lrRecord.responseReceivedTime).toStrictEqual(7.5);
+      expect(lrRecord.timing.proxyStart).toStrictEqual(-1);
+      expect(lrRecord.timing.connectStart).toStrictEqual(0);
+      expect(lrRecord.timing.connectEnd).toStrictEqual(5000);
+      expect(lrRecord.timing.sslStart).toStrictEqual(4000);
+      expect(lrRecord.timing.sslEnd).toStrictEqual(5000);
+      expect(lrRecord.timing.sendStart).toStrictEqual(5000);
+      expect(lrRecord.timing.sendEnd).toStrictEqual(5000);
+      expect(lrRecord.timing.receiveHeadersEnd).toStrictEqual(7500);
+      expect(lrRecord.lrStatistics).toStrictEqual({
+        endTimeDeltaMs: -8000,
+        TCPMs: 5000,
+        requestMs: 2500,
+        responseMs: 2500,
+      });
+    });
+
+    it('merges with existing timing properties', function() {
```

Comment:
is there any situation where we'll create inconsistent timings here? seems like we probably *don't* want to merge the timings right?

Reply:
then again I imagine this won't be a real-world scenario right?

Reply:
Hm, interesting thought.  @paulirish thoughts?  I think this is moot since we expect it to never happen, but we may want to just override everything.

Reply:
yup lets just override. seems like we can just drop the `if` statement here

https://github.com/GoogleChrome/lighthouse/blob/a0e5fd24f95a166c3495a13f0e2ff5a5f9e907ef/lighthouse-core/lib/network-request.js#L432


cool

Reply:
Done.